### PR TITLE
Fix UnsupportedOperationException in BugRanker.trimToMaxRank

### LINKED ISSUE #1161: UnsupportedOperationException after upgrade from 3.1.12 to 4.0.4
Hello,

I have following problem with SpotBugs:

In my project's version 3.1.12 is running fine - we are using it from our Jenkins/Ant scripts.
After changing it to 4.0.4 at some step of our process following exception is thrown in computeBugHistory:

[computeBugHistory] Executing SpotBugs ComputeBugHistoryTask from ant task
[computeBugHistory] Running computeBugHistory...
[computeBugHistory] Starting with d:\\spotbugs-4.0.4/database.xml
[computeBugHistory] Exception in thread "main" java.lang.UnsupportedOperationException
[computeBugHistory]     at java.util.Collections$UnmodifiableCollection.removeIf(Collections.java:1084)
[computeBugHistory]     at edu.umd.cs.findbugs.BugRanker.trimToMaxRank(BugRanker.java:322)
[computeBugHistory]     at edu.umd.cs.findbugs.workflow.Update.discardUnwantedBugs(Update.java:337)
[computeBugHistory]     at edu.umd.cs.findbugs.workflow.Update.doit(Update.java:484)
[computeBugHistory]     at edu.umd.cs.findbugs.workflow.Update.main(Update.java:405)
[computeBugHistory] Java Result: 1

I checked quickly source code of BugRanker and Update classes and it seems there is call to removeIf() method of unmodifiable view of collection - and that lead to UnsupportedOperationException.

Is it bug or some additional steps are needed when upgrade from 3 to 4?

Attaching script and data file (truncated, as simple as possible to reproduce bug).
[spotbugs.zip](https://github.com/spotbugs/spotbugs/files/4758854/spotbugs.zip)

### PR DESCRIPTION
**Pull request by aajisaka**
Use `SortedBugCollection.iterator()` to get a mutable iterator instead of immutable `SortedBugCollection.getCollection()`.

Fixes #1161 

----

Make sure these boxes are checked before submitting your PR -- thank you!

- [x] Added an entry into `CHANGELOG.md` if you have changed SpotBugs code

### CODE REVIEW

File: spotbugs/src/test/java/edu/umd/cs/findbugs/BugRankerTest.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package edu.umd.cs.findbugs;
+
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+
+public class BugRankerTest {
+    /**
+     * @see <a href="https://github.com/spotbugs/spotbugs/issues/1161"></a>
```
Comment (KengoTODA) [inline spotbugs/src/test/java/edu/umd/cs/findbugs/BugRankerTest.java:9]: add a text to link, or hyperlink won't work.

**Review by KengoTODA — CHANGES_REQUESTED** [review]
Cool, just a minor change we need. Thanks for your contribution!

File: spotbugs/src/test/java/edu/umd/cs/findbugs/BugRankerTest.java (hunk 1 above)
Comment (aajisaka) [reply spotbugs/src/test/java/edu/umd/cs/findbugs/BugRankerTest.java]: Thank you for your review. Added a text https://github.com/spotbugs/spotbugs/pull/1436/commits/913f233b9bd1f8ec22fedd8d8cafd1ee489d9c35

**Review by KengoTODA — APPROVED** [review]

**Comment by aajisaka** [comment]
Thank you @KengoTODA!

**Pull request merged**